### ISSUE TITLE
Latest update: Path of Building no longer launches in <Lutris (5.7-11) Wine runner

### ISSUE THREAD

**Issue by piv0tal** [PathOfBuildingCommunity/PathOfBuilding#1678]
After updating the latest PoB version, launching the program attempts to perform the standard green-screen update (quick pop up), then closes. No further execution takes place. Program no longer launches.

System: Ubuntu 20.04.1
Lutris 5.7-11x86

Tested various older versions of wine - same result

DXVK on or off, same result

**Comment by hplato** [comment]
Same on macos wine.

wine: Unhandled page fault on read access to 0x00000000 at address 0x57810f (thread 0009), starting debugger...
Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x0057810f).

System information:
    Wine build: wine-4.0
    Platform: i386 (WOW64)
    Version: Windows 7
    Host system: Darwin
    Host version: 18.7.0

**Comment by codestation** [comment]
Same on the latest Wine version, the backtrace points to

```
=>0 0x00ad810f EntryPoint+0xffffffff() in simplegraphic (0x0031e440)
```

Wine build: wine-5.19 (Staging)

**Comment by dclamage** [comment]
I'm going to have to install Linux and Wine to reproduce this for a fix, but in the meantime can you guys try grabbing the old simplegraphic.dll and use that in your installation directory and see if that fixed the issue?

**Comment by JambonII** [comment]
> I'm going to have to install Linux and Wine to reproduce this for a fix, but in the meantime can you guys try grabbing the old simplegraphic.dll and use that in your installation directory and see if that fixed the issue?

I try to use simplegraphic.dll ( from 1.4.170.12 )
Run POB ( fresh install from .exe )  => auto-update => crash => copy simplegraphic.dll again => run POB => It's work !

**Comment by hplato** [comment]
Same with me, replacing simplegraphic.dll from older version, and 1.4.170.16 starts fine. However I think it's still downloading the newer simplegraphic.dll as I see it's checking for a new version, downloads 1 file and then says update ready.

**Comment by tpdns90321** [comment]
1. Download 'Old' release pob portable zip.
2. unzip and copy SimpleGraphic.dll to your using pob's folder
3. Run

Thanks! @JambonII

**Comment by piv0tal** [comment]
Can confirm copying/pasting the current PoB install file SimpleGraphic.dll to my existing installation fixed it --

Now I'm scared to update it again

**Comment by Sanatorius** [comment]
I have a similar problem on Windows 10. I start the update, a window called "UpdateApply.lua" opens and shows "Applying update... ; next line: Updating 'path\Path of Building Community/SimpleGraphic.dll' ;" and nothing else happens. after i close the window and start pob again, i get the message to update. so the update doesnt work.

![image](https://user-images.githubusercontent.com/24896350/97312014-50207500-1865-11eb-857c-62b3008e399f.png)

![image](https://user-images.githubusercontent.com/24896350/97312231-8b22a880-1865-11eb-9337-38464230f4ad.png)

**Comment by mpoquet** [comment]
Can reproduce error on NixOS.

```
System information:
    Wine build: wine-4.0.2
    Platform: i386
    Version: Windows 7
    Host system: Linux
    Host version: 4.14.148
```

## Workaround
Using the `SimpleGraphic.dll` released with `1.4.170.2` *works*, but as the file is replaced by (forced and optional) updates, it needs to be manually replaced over and over.

Here is a quick script to obtain a working PoB-1.4.170.16 setup.
**Ignore non forced update proposals or you'll need to replace the library once again**.

``` bash
# retrieve versions
curl -kL -o ./PoB-old.zip https://github.com/PathOfBuildingCommunity/PathOfBuilding/releases/download/v1.4.170.2/PathOfBuildingCommunity-Setup-1.4.170.2.zip
curl -kL -o ./PoB-new.zip https://github.com/PathOfBuildingCommunity/PathOfBuilding/releases/download/v1.4.170.16/PathOfBuildingCommunity-Setup-1.4.170.16.zip
mkdir ./PoB-old && (cd ./PoB-old && unzip ../PoB-old.zip)
mkdir ./PoB-new && (cd ./PoB-new && unzip ../PoB-new.zip)

# replace library (1)
cp ./PoB-old/SimpleGraphic.dll ./PoB-new/

# first run
# - should start without crash (thanks to library replacement)
# - should crash during automatic forced update
wine './PoB-new/Path of Building.exe'

# replace library (2)
cp ./PoB-old/SimpleGraphic.dll ./PoB-new/

# second run
# - should start without crashing (thanks to library replacement)
wine './PoB-new/Path of Building.exe'
```

## Fix proposals
- Quick: rollback `SimpleGraphic.dll`, release a new PoB with working lib.
- Best: fix `SimpleGraphic.dll`, release a new PoB with working lib.

## First update discussion
Forcing the first update seems like a bad/dangerous idea to me.

Users may decide to use an older version for some reason, and I feel like they should be able to do so easily (why not with big warning popups if some breaking changes occurred).

The dangerous part is that if in the future, recent PoB releases crash on some environments, users will not be able to use previous PoB releases as safe workarounds, which is quite sad as previous versions worked perfectly on their environment.

(It also feels quite strange to me to have a forced update just after downloading a 30+ MB archive.)

**Comment by ryannathans** [comment]
If you wanna avoid the updates downloading and breaking the work around, set the proxy server address to something that isn't a proxy - eg 127.0.0.1 port 12345

**Comment by tainium** [comment]
Just chiming in as another user with this problem.  I've tried other workarounds and different workstations but the fix in this thread is the only way to get it working.

**Comment by Wires77** [comment]
Okay, I believe I have a fixed version of the DLL ready now, can some of you download the DLL from my branch here and see if it works for you? https://github.com/Wires77/PathOfBuilding-SimpleGraphic/runs/1478065877?check_suite_focus=true

I tested this on my Ubuntu VM and it seemed to work fine.

**Comment by tainium** [comment]
@Wires77 That works!

What I did:
Loaded 1.4.170.11
Auto-updated
Crashed
Pasted new DLL from @Wires77 over the old DLL
Loaded 1.4.170.18 w/new DLL...Success!
It says update is available due to the DLL...crashes if I apply that update, of course

**Comment by codestation** [comment]
Same, updated to 1.4.170.18 and replaced DLL, it works while i don't apply more updates. The only thing left to do would be to test that it continues to work on Windows.

**Comment by Wires77** [comment]
This fix will go into the next release of PoB.  If anyone is curious, I've filed a bug report with Wine here: https://bugs.winehq.org/show_bug.cgi?id=50232

**Comment by LocalIdentity** [comment]
Fixed in 1.4.170.20
